### PR TITLE
Migrate to Null Safety

### PR DESCRIPTION
**Pull request by Fox32**
This migrates the package to null safety based on [this guide](https://dart.dev/null-safety/migration-guide). I also removed the pubspec.lock based on [these docs](https://dart.dev/guides/libraries/private-files).

I'm a bit unsure about the migration around the knobs, especially the `options` one. I kept it to be nullable as otherwise the initial value has to be required.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: '>=2.12.0-133.2.beta <3.0.0'
 
 dependencies:
-  provider: ">=4.0.0 <5.0.0"
-  recase: ">=2.0.1 <4.0.0"
-  rxdart: ">=0.24.0 <1.0.0"
+  collection: ^1.15.0-nullsafety.5
```
Comment (Fox32) [inline pubspec.yaml:10]: `collection` is a well tested and stable package, I think it's better than writing the code for `groupBy` yourself. Also the migration tool suggested to use it.

File: lib/src/knobs/knobs.dart [hunk 2]
```diff
@@ -42,33 +42,34 @@ abstract class KnobsBuilder {
   });
 
   /// Creates select field with [label], [initial] value and list of [options].
-  T options<T>(String label, {T initial, List<Option<T>> options = const []});
+  T? options<T>(String label, {T? initial, List<Option<T>> options = const []});
```
Comment (ookami-kb) [inline lib/src/knobs/knobs.dart:45]: Let's rather make it as `T` and use it without `?` and `!` everywhere.

First, using `!` for me is actually a code smell. Second, I'd say it's for the user to decide, whether they want nullable or non-nullable options (i.e. in your example with `MouseCursor`, the user should be free to choose `options<MouseCursor>` or `options<MouseCursor?>`).

**Review by ookami-kb — CHANGES_REQUESTED** [review]
Thanks for your PR!

Just a couple of comments, otherwise LGTM.

File: lib/src/knobs/knobs.dart (hunk 2 above)
Comment (Fox32) [reply lib/src/knobs/knobs.dart]: I updated the PR, have a look.

That makes the `initial` a required value, but I think that's fine. If you really want to use `null` then you have to pass it as an `initial` value. See the example for that. 

Now I wonder, if `options()` has required named parameters, should we change the API to only use named parameters? Like:

```dart
  T? options<T>({required String label, required T initial, List<Option<T>> options = const []});
```

That should be done for `boolean` and `text` too. What do you think? But that could also be part of a separate PR.

Comment (ookami-kb) [reply lib/src/knobs/knobs.dart]: Yeah, it makes sense to make it named as well. I'll merge this PR and will make it separately.

**Review by ookami-kb — APPROVED** [review]

**Comment by Fox32** [comment]
Cool, thanks!

**Pull request merged**